funcx_endpoint/requirements.txt

```diff
@@ -1,17 +1,45 @@
-requests>=2.20.0
-jsonschema>=3.2.0
-configobj
-texttable
-psutil
-python-daemon
-fair_research_login
+requests>=2.20.0,<3
 globus_sdk<3
-dill>=0.3
-typer>=0.3.0
 funcx>=0.3.3,<0.4.0
-# disallow use of 22.3.0
-# whl packages on some platforms seem to produce ZMQ issues
-# FIXME: re-evaluate this and see if the version pin can be removed after the
-# forwarder is upgraded to a newer libzmq
+
+# table printing used in list-endpoints
+# for detail on the bound, see `funcx_sdk/requirements.txt`
+texttable==1.6.4
+
+# although psutil does not declare itself to use semver, it appears to offer
+# strong backwards-compatibility promises based on its changelog, usage, and
+# history
+#
+# TODO: re-evaluate bound after we have an answer of some kind from psutil
+# see:
+#   https://github.com/giampaolo/psutil/issues/2002
+psutil<6
+
+# provides easy daemonization of the endpoint
+python-daemon>=2,<3
+
+# TODO: replace use of `typer` with `click` because
+# 1. `typer` is a thin wrapper over `click` offering very minimal additional
+#    functionality
+# 2. `click` follows semver and releases new major versions when known
+#    backwards-incompatible changes are introduced, making our application
+#    safer to distribute
+typer==0.4.0
+
+
+# disallow use of 22.3.0; the whl package on some platforms causes ZMQ issues
+#
+# NOTE: 22.3.0 introduced a patched version of libzmq.so to the wheel packaging
+# which may be the source of the problems , the problem can be fixed by
+# building from source, which may mean there's an issue in the packaged library
+# further investigation may be needed if the issue persists in the next pyzmq
+# release
 pyzmq>=22.0.0,!=22.3.0
-retry
+
+# TODO: evaluate removal of the 'retry' library after the update to
+# globus-sdk v3, which provides automatic retries on all API calls
+retry==0.9.2
+
+# 'parsl' is a core requirement of the funcx-endpoint, essential to a range
+# of different features and functions
+parsl>=1.1.0,<2
```

Comment:
see equivalent comment in #613 review - parsl is not semver and the version pin should be `==`